nginx_stage/lib/nginx_stage.rb

```diff
@@ -141,8 +141,13 @@ def self.nginx_env(user:)
       "OOD_FILES_URL" => "/pun/sys/dashboard/files",
       # this is not a typo => the editor is /edit off of the base url
       "OOD_EDITOR_URL" => "/pun/sys/dashboard/files",
-      "RAILS_LOG_TO_STDOUT" => "true"
-    }.merge(pun_custom_env)
+      "RAILS_LOG_TO_STDOUT" => "true",
+      "PATH" => ENV['PATH'],
+      "LD_LIBRARY_PATH" => ENV['LD_LIBRARY_PATH'],
+      "RUBYLIB" => ENV['RUBYLIB'],
+      "GEM_HOME" => ENV['GEM_HOME'],
+      "GEM_PATH" => ENV['GEM_PATH'],
+    }.merge(pun_custom_env))
```

Comment:
Instead of just merging with `pun_custom_env` we likely need to merge with `env_declarations` too if we're clearing the env.

```suggestion
    }.merge(pun_custom_env).merge(env_declarations.map { |k| [ k, ENV[k] ] }.to_h))
```